spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/rdd/RDDSpec.scala

```diff
@@ -201,6 +201,38 @@ class RDDSpec extends SparkCassandraITFlatSpecBase {
     checkArrayCassandraRow(result)
   }
 
+  it should "throw a meaningful exception if partition column is null when joining with Cassandra table" in {
+    val source = sc.parallelize(keys).map(x ⇒ new KVWithOptionRow(None))
+    try {
+      source.joinWithCassandraTable[(Int, Long, String)](ks, tableName).collect()
+      fail()
```

Comment:
I'm not clear why you are manually adding fail() shouldn't you be testing args that are null/None and the changed code you wrote be throwing the failure to note the args were not valid? In which case you should be doing

intercept[IllegalArgumentException] {
  source.doThingsThatShouldFail()
}